### DISCUSSION TITLE
What stream security tokens are best for restricting only logged in users to play/publish live streams?

### DISCUSSION THREAD

**Discussion by tremzo** [ant-media/Ant-Media-Server#5360] [FAQs]
There are a few different stream security tokens available, JWT, TOTP and One-Time tokens that can be used to secure the publishing and playing back of a live stream. 

In the case of restricting the play/publish of live streams to only users that have logged into the platform, which type of token is recommended?

This blog post suggests using TOTP to enable the creation, deletion and management of subscribers https://antmedia.io/time-based-one-time-password-totp-makes-streams-secure/.

**Comment by USAMAWIZARD** [comment]
you can implement something like this.
1. on your backend side you will have to track the sessions of the logged in users.
2. user sends a request for playing or publishing the video to your backend 
3. your back end will verify is user is logged and and session is active is yes 
4. you can send request from back end to AMS  to generate [one time token](https://antmedia.io/docs/guides/developer-sdk-and-api/rest-api-guide/stream-security/#2-one-time-token-control) for either play or publish according to user request using [REST API](https://antmedia.io/rest/#/BroadcastRestService/getTokenV2) . 
5. once you get the generated one time token send the play / publish page to user with token and now he / she can publish or play  the video
